src/aggregator/client/queue.go

```diff
@@ -193,10 +192,16 @@ func (q *queue) Enqueue(buf protobuf.Buffer) error {
 		default:
 			return errInvalidDropType
 		}
+
+		full = q.buf.full()
 	}
 
+	// NB: The qbuf can still hold a single super huge buffer way bigger
+	// maxSizeBytes.
 	q.buf.push(buf)
 	q.metrics.enqueueSuccesses.Inc(1)
+	q.metrics.queueLen.RecordValue(float64(q.buf.getSizeBytes()))
+	q.metrics.queueBufSizes.RecordValue(float64(len(buf.Bytes())))
```

Comment:
this only makes sense in Flush or periodic sampling context - otherwise you'll be recording ever-incrementing counts that will be hard to make sense of. what you really want is a way to compute the P99 etc. buffer size at a point in time (say, over 10 seconds), ideally when the buffer is at its largest.